addons/web/static/src/legacy/js/views/kanban/kanban_record.js

```diff
@@ -610,31 +610,35 @@ var KanbanRecord = Widget.extend(WidgetAdapterMixin, {
      * @returns {Object} transformed record data
      */
     _transformRecord: function (recordData) {
-        var self = this;
-        var new_record = {};
-        _.each(this.state.getFieldNames(), function (name) {
-            var value = recordData[name];
-            var r = _.clone(self.fields[name] || {});
+        const new_record = Object.create(null);
+        for (const name of this.state.getFieldNames()) {
+            const value = recordData[name];
+            const r = new_record[name] = _.clone(this.fields[name] || {});
 
-            if ((r.type === 'date' || r.type === 'datetime') && value) {
-                r.raw_value = value.toDate();
-            } else if (r.type === 'one2many' || r.type === 'many2many') {
+            r.raw_value = value;
+            switch (r.type) {
+            case 'date': case 'datetime':
+                r.raw_value = value && value.toDate();
+                break;
+            case 'one2many': case 'many2many':
                 r.raw_value = value.count ? value.res_ids : [];
-            } else if (r.type === 'many2one') {
+                break;
+            case 'many2one':
                 r.raw_value = value && value.res_id || false;
-            } else {
-                r.raw_value = value;
+                break;
             }
 
-            if (r.type) {
-                var formatter = field_utils.format[r.type];
-                r.value = formatter(value, self.fields[name], recordData, self.state);
-            } else {
+            switch (r.type) {
+            case 'html': // or should this be in `format` instead of _.identity?
+                r.value = utils.Markup(value);
+                break;
+            case '': case null: case undefined:
```

Comment:
how could the field type be one of these?

Reply:
I don't know, I just converted from the existing conditional of
```javascript
if (r.type) {
    ...
} else {
    ...
}
```
as in the `switch` the generic `r.type` becomes the default case.

Reply:
weird, but ok, as you say, it was like this before (except that before we checked that the value was falsy, you're code doesn't handle the case where the field type is 0 :D)

It's maybe worth trying to remove it and see what breaks

Reply:
> weird, but ok, as you say, it was like this before (except that before we checked that the value was falsy, you're code doesn't handle the case where the field type is 0 :D)

Only for the `date` field, and it would fall through to `r.raw_value = value` so I just did that using a single conditional.

Reply:
> It's maybe worth trying to remove it and see what breaks

Yeah I'm going to, if it works fine then this simplifies the conditions a lot and merging the two `switch` is easy.

In fact I'm thinking it could work by just defaulting the `formatter` to an identity function:
```js
            const formatter = field_utils.format[r.type] || _.identity;
```
since the falsy types are not present in the format map (duh) the value will just go through, and then we can directly assign to `value` instead of doing so conditionally:
```js
            r.raw_value = value;
            r.value = formatter(value, this.fields[name], recordData, this.state);
```
and the `switch` just handles actual divergences / overwrites but don't need to repeat the base case.

The only possible concern is that in valid field types will now go through instead of trying to call `undefined`, but that seems like a rather limited risk for "legacy" formatting. If anything it's even useful as it means new field types will "work" by default instead of triggering errors, so not the worst.